src/aot/aot.cpp

```diff
@@ -253,16 +253,66 @@ int load(BPFtrace &bpftrace, const std::string &in)
     return 1;
   }
 
-  uint8_t *ptr = static_cast<uint8_t *>(
-      ::mmap(NULL, in_file_size, PROT_READ, MAP_PRIVATE, infd, 0));
-  if (ptr == MAP_FAILED) {
-    auto saved_err = errno;
-    LOG(ERROR) << "Failed to mmap: " << in << ": " << std::strerror(saved_err);
-    return 1;
+  // Find .btaot section
+  Elf *elf = NULL;
+  Elf_Scn *scn = NULL;
+  GElf_Shdr shdr;
+  char *secname = NULL;
+  Elf_Data *data = NULL;
+  uint8_t *btaot_section = NULL;
+  const Header *hdr;
+
+  if (elf_version(EV_CURRENT) == EV_NONE) {
+    LOG(ERROR) << "Cannot set libelf version: " << elf_errmsg(-1);
+    goto out;
+  }
+
+  elf = elf_begin(infd, ELF_C_READ, NULL);
+  if (!elf) {
+    LOG(ERROR) << "Cannot read ELF file: " << elf_errmsg(-1);
+    goto out;
   }
 
-  // Validate header
-  auto hdr = reinterpret_cast<const Header *>(ptr);
+  size_t strndx;
+  if (elf_getshdrstrndx(elf, &strndx) < 0)
+    goto out;
+
+  int i;
+  i = 0;
```

Comment:
super nit
```suggestion
  int i = 0;
```

Reply:
Unfortunately, because of the three goto statements above this line, doing ```int i = 0;``` causes the compiler to complain with ```error: jump to label ‘out .... ’crosses initialization of ‘int i’```. We can work around it by splitting the declaration and assignment like so, or moving the whole thing above the gotos (which it was in a previous version of the commit, but I think we prefer to have ```int i``` closer to the while loop).

Reply:
Huh, that is surprising but ok.